compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/nodes/ArrayFillNode.java

```diff
@@ -0,0 +1,202 @@
+ */
+@NodeInfo(cycles = NodeCycles.CYCLES_UNKNOWN, size = NodeSize.SIZE_UNKNOWN, allowedUsageTypes = InputType.Memory)
+public class ArrayFillNode extends MemoryKillStubIntrinsicNode
+                implements StateSplit, MemoryAccess {
+
+    public static final NodeClass<ArrayFillNode> TYPE = NodeClass.create(ArrayFillNode.class);
+    private static final ForeignCallDescriptor BOOLEAN_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Boolean);
+    private static final ForeignCallDescriptor BYTE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Byte);
+    private static final ForeignCallDescriptor CHAR_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Char);
+    private static final ForeignCallDescriptor SHORT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Short);
+    private static final ForeignCallDescriptor INT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Int);
+    private static final ForeignCallDescriptor LONG_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Long);
+    private static final ForeignCallDescriptor FLOAT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Float);
+    private static final ForeignCallDescriptor DOUBLE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Double);
+
+    public static final ForeignCallDescriptor[] STUBS = {
+                    BOOLEAN_ARRAY_FILL_STUB,
+                    BYTE_ARRAY_FILL_STUB,
+                    CHAR_ARRAY_FILL_STUB,
+                    SHORT_ARRAY_FILL_STUB,
+                    INT_ARRAY_FILL_STUB,
+                    LONG_ARRAY_FILL_STUB,
+                    FLOAT_ARRAY_FILL_STUB,
+                    DOUBLE_ARRAY_FILL_STUB,
+    };
+
+    protected JavaKind elementKind;
+
+    @Input protected ValueNode arrayBase;
+    @Input protected ValueNode offsetToFirstElement;
+    @Input protected ValueNode arrayLength;
+    @Input protected ValueNode valueToFillWith;
+
+    public ArrayFillNode(ValueNode arrayBase, ValueNode offsetToFirstElement, ValueNode arrayLength, ValueNode valueToFillWith, @ConstantNodeParameter JavaKind elementKind) {
+        super(TYPE, StampFactory.forVoid(), null, LocationIdentity.any());
+        this.arrayBase = arrayBase;
+        this.offsetToFirstElement = offsetToFirstElement;
+        this.arrayLength = arrayLength;
+        this.valueToFillWith = valueToFillWith;
+        this.elementKind = elementKind;
+    }
+
+    public ArrayFillNode(ValueNode arrayBase, ValueNode offsetToFirstElement, ValueNode arrayLength, ValueNode valueToFillWith, @ConstantNodeParameter JavaKind elementKind,
+                    @ConstantNodeParameter EnumSet<?> runtimeCheckedCPUFeatures) {
+        super(TYPE, StampFactory.forVoid(), runtimeCheckedCPUFeatures, LocationIdentity.any());
+        this.arrayBase = arrayBase;
+        this.offsetToFirstElement = offsetToFirstElement;
+        this.arrayLength = arrayLength;
+        this.valueToFillWith = valueToFillWith;
+        this.elementKind = elementKind;
```

Comment:
Please add guarantees for:
- `valueToFillWith.stamp(NodeView.DEFAULT) instanceof IntegerStamp`
- `elementKind.isPrimitive()`
- `PrimitiveStamp.getBits(valueToFillWith.stamp(NodeView.DEFAULT)) == elementKind.getBitCount()`